bpf/flows.c

```diff
@@ -80,7 +80,7 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
     id.direction = direction;
 
     // check if this packet need to be filtered if filtering feature is enabled
-    bool skip = check_and_do_flow_filtering(&id, pkt.flags, 0);
+    bool skip = check_and_do_flow_filtering(&pkt, 0);
```

Comment:
I don't understand the need to replace id with pkt struct that generated lots of unnecessary changes and I also planned to even delete the pkt info struct at some point ?

Reply:
this is due to removing stuff from flow_id, and moving them to flow_metrics, like MAC addresses. As a result, id isn't sufficient to pass it to all these functions in `utils`. So instead of adding tons of arguments to the signatures I just use the `pkt` that already has everything needed

Reply:
`pkt` is convenient to not have extra long signatures

Reply:
u just removed l2 and interface right ? or am I missing something else ? why would flow filter or util core apis need to change ?

Reply:
and `eth_protocol` and `direction`

Reply:
remember datapath has to be efficient adding additional dereferrence isn't efficient IMO that stuff not used and I don't think will impact the apis to change the signature much

Reply:
I'm reaching the 5 args limit of ebpf calls if I want to remove them from `pkt`

Reply:
fixed via https://github.com/netobserv/netobserv-ebpf-agent/pull/469/commits/8e69cfd4a3703561ac26b8218d4079aa11e25edc 
that allows also to get rid of a few unnecessary MAC copying